modules/geo-layers/src/tile-layer/utils/viewport-util.js

```diff
@@ -18,10 +25,10 @@ function getBoundingBox(viewport) {
   ];
 }
 
-function getTileIndex(lngLat, scale) {
-  let [x, y] = lngLatToWorld(lngLat);
-  x *= scale / TILE_SIZE;
-  y = (1 - y / TILE_SIZE) * scale;
+function getTileIndex(lngLat, scale, tileSize) {
+  let [x, y] = lngLatToWorld(lngLat[0], lngLat[1], tileSize);
+  x *= scale / tileSize;
+  y = (1 - y / tileSize) * scale;
```

Comment:
`tileSize` does not matter here because you removed it right away. The old `getTileIndex` implementation should return the same result.

Reply:
Thanks, the math definitely is redundant.  Should we just hardcode the `tileSize` prop for `TileLayer` then to be 512 to match the `math.gl` [implementation](https://github.com/uber-web/math.gl/blob/master/modules/web-mercator/src/web-mercator-utils.js#L15) even though it could be incorrect for the actual tiles?  I don't think the parameter is used anywhere else in the `geo-layers` `TileLayer` implementation so being "incorrect" won't have any real impact. I can also leave a comment.

Reply:
Yeah I think hard coding it would be fine since it's just for this layer. `512` is needed by the `WebMercatorViewport` to match Mapbox projection.

Ultimately, `tileSize` is not needed by any of the public APIs, because `getTileIndices` and `getTileBoundingBox` would effectively determine the size of the tiles.